### ISSUE TITLE
One OIDC provider and two IAM roles

### ISSUE THREAD

**Issue by aleksei-zolotukhin** [saidsef/terraform-aws-terraform-cloud-oidc#20]
### Description of Problem:
Terraform Cloud allows to specify `TFC_AWS_PLAN_ROLE_ARN` and `TFC_AWS_APPLY_ROLE_ARN` for using different IAM roles for different stages.

I thought of using the module for creating two definitions:

- The first creates OIDC provider for TFC and IAM role with ReadOnly permissions
- The second does not create the OIDC provider and uses the created in the previous step, but also creates IAM role with AdministratorAccess permissions

```
#
# Define OIDC provider for TFC and ReadOnly IAM role
#
module "tfc_oidc_provider_and_readonly_role" {
  source  = "saidsef/terraform-cloud-oidc/aws"
  version = "1.7.3"

  attach_admin_policy           = false
  attach_read_only_policy       = true
  
  create_oidc_provider          = true
  enabled                       = true
  
  force_detach_policies         = false
  organisation                  = var.tfc_organization_name
  projects                      = local.tfc_projects
  iam_role_name                 = "terraform-cloud-readonly-iam-role-${local.account_id}"
  iam_role_path                 = "/"
  iam_role_permissions_boundary = ""
  iam_role_policy_arns          = []
  max_session_duration          = 3600
  tags                          = local.tags
  url                           = var.tfc_hostname
}

#
# Define IAM role with admin permissions for existing OIDC provider
#
module "tfc_oidc_provider_admin_role" {
  source  = "saidsef/terraform-cloud-oidc/aws"
  version = "1.7.3"
  depends_on = [
    module.tfc_oidc_provider_and_readonly_role
  ]

  attach_admin_policy           = true
  attach_read_only_policy       = false
  
  create_oidc_provider          = false
  enabled                       = true
  
  force_detach_policies         = false
  organisation                  = var.tfc_organization_name
  projects                      = local.tfc_projects
  iam_role_name                 = "terraform-cloud-admin-iam-role-${local.account_id}"
  iam_role_path                 = "/"
  iam_role_permissions_boundary = ""
  iam_role_policy_arns          = []
  max_session_duration          = 3600
  tags                          = local.tags
  url                           = var.tfc_hostname
}
```
However, when I run the `plan` command I receive the following error:
```
╷
│ Error: Invalid index
│ 
│   on .terraform/modules/tfc_oidc_provider_admin_role/current.tf line 22, in data "aws_iam_policy_document" "assume_role":
│   22:       values   = [format("%s", one(aws_iam_openid_connect_provider.provider[0].client_id_list))]
│     ├────────────────
│     │ aws_iam_openid_connect_provider.provider is empty tuple
│ 
│ The given key does not identify an element in this collection value: the collection has no elements.
╵
```

### Potential Solutions:
I suppose, there might be added a parameter of the module containing the reference to previously defined OIDC provider.

**Comment by saidsef** [comment]
Hi @aleksei-zolotukhin , thanks for reporting the issue.

There is actually a local variable to to control if the the module should reuse existing module or create new module:

https://github.com/saidsef/terraform-aws-terraform-cloud-oidc/blob/0409a10dafab1eecfbba3e25e3fd66146ae85aa5/main.tf#L2

However it is **NOT** being used, I will create PR fix that and create a new tag after testing.

**Comment by saidsef** [comment]
Hi @aleksei-zolotukhin, that should be fixed in [v1.7.4](https://github.com/saidsef/terraform-aws-terraform-cloud-oidc/releases/tag/v1.7.4) tag release

**Comment by aleksei-zolotukhin** [comment]
Hi @saidsef, I have tested your changes recently and now it works as intended.
I really appreciate for your responsiveness and efforts for fixing the bug!
